File: ruby/hamming/hamming.rb

class Hamming
  VERSION = 1

  def self.compute(left, right)
    if left.length != right.length
      raise ArgumentError
    end
    count = 0
    left.chars.each_index do |index|
      count += 1 unless left[index] == right[index]
    end
    count
  end
end

File: java/hamming/src/main/java/Hamming.java
public class Hamming {
    private String leftStrand;
    private String rightStrand;

    Hamming(String leftStrand, String rightStrand) {
        if (leftStrand.length() != rightStrand.length()) {
            throw new IllegalArgumentException("leftStrand and rightStrand must be of equal length.");
        } else {
            this.leftStrand = leftStrand;
            this.rightStrand = rightStrand;
        }
    }

    int getHammingDistance() {
        int count = 0;
        for (int index = 0; index < leftStrand.length(); index++) {
            if (leftStrand.charAt(index) != rightStrand.charAt(index)) {
                count += 1;
            }
        }
        return count;
    }
}
